# **Hypothesis Testing**

## Objectives

- Perform hypothesis testing to examine relationships between key variables and student dropout.

## Inputs

- Cleaned student dataset `clean_student_dropout_dataset.csv`

## Outputs

- Statistical test results (Pearson correlation, t-test and Chi-square)
- Interpretation of each hypothesis

## Additional Comments

- Only basic statistical methods taught in the course were used.



---

In [31]:
import pandas as pd
import scipy.stats as stats

In [32]:
df = pd.read_csv("../Dataset/Processed/clean_student_dropout_dataset.csv")
df.head()

,Student_ID,Age,Gender,Family_Income,Internet_Access,Study_Hours_per_Day,Attendance_Rate,Assignment_Delay_Days,Travel_Time_Minutes,Part_Time_Job,Scholarship,Stress_Index,GPA,Semester_GPA,CGPA,Semester,Department,Parental_Education,Dropout
0,1,22.1,0,25000.0,1,3.36,86.1,2,20.4,1,0,5.5,0.96,0.90,0.90,Year 1,Arts,High School,0
1,2,20.7,0,25000.0,1,4.30,68.0,2,44.0,0,0,6.8,1.28,1.20,1.19,Year 3,Engineering,Bachelor,1
2,3,22.4,0,40183.0,1,4.40,70.9,0,48.9,1,0,5.5,1.68,1.32,1.32,Year 1,Arts,Master,0
3,5,20.5,1,25319.0,1,4.19,75.7,1,23.0,0,0,7.0,1.48,0.91,0.87,Year 4,Business,Bachelor,0
4,7,24.5,0,25000.0,1,3.00,78.2,1,37.4,1,1,7.3,0.64,0.33,0.44,Year 4,CS,Bachelor,0


---

### Hypothesis 1 – Attendance Rate and Dropout

H₀: There is no relationship between attendance rate and student dropout.

H₁: There is a relationship between attendance rate and student dropout.

Test: Pearson Correlation

In the context of the dataset: The analysis will examine whether students with lower attendance rates are more likely to drop out.

In [33]:
attendance = df["Attendance_Rate"]
dropout = df["Dropout"]

In [34]:
corr, p_value = pearsonr(attendance, dropout)

In [35]:
print("Correlation coefficient:", corr)
print("P-value:", p_value)

Correlation coefficient: -0.1729746135396005
P-value: 1.5836249047684234e-61


#### What this code does

- The variables Attendance_Rate and Dropout were selected from the dataset.

- A Pearson correlation test was performed to measure the relationship between attendance rate and dropout.

- The test outputs a correlation coefficient and a p-value to show the strength and significance of the relationship.

### Interpretation

- The test produced a correlation coefficient of -0.173 and a p-value of 1.58e-61.

- Since the p-value is less than 0.05, the null hypothesis is rejected.

- This indicates a statistically significant relationship, where higher attendance is associated with lower dropout.

### In Context of the Dataset

- The results suggest that students with lower attendance are more likely to drop out.

- Improving attendance may help reduce dropout rates.

---

### Hypothesis 2 – Gender and Dropout

H₀: There is no significant difference in dropout rates between male and female students.

H₁: There is a significant difference in dropout rates between male and female students.

Test: Independent t-test

In the context of the dataset: The analysis will compare dropout rates between male and female students to determine if gender is associated with student dropout.

In [36]:
male_dropout = df[df["Gender"] == 0]["Dropout"].dropna()
female_dropout = df[df["Gender"] == 1]["Dropout"].dropna()

In [37]:
t_stat, p_value = stats.ttest_ind(male_dropout, female_dropout)

In [38]:
print("T-statistic:", t_stat)
print("P-value:", p_value)

T-statistic: -0.5697596119064169
P-value: 0.5688549445340554


#### What this code does

- The dataset was divided into two groups based on gender (0 = male, 1 = female).

- An independent samples t-test was performed to compare dropout rates between male and female students.

- The test outputs a t-statistic and a p-value to determine whether the difference between the two groups is statistically significant.

### Interpretation

- The test produced a t-statistic of -0.57 and a p-value of 0.569.

- Since the p-value is greater than 0.05, the null hypothesis is not rejected.

- This indicates that there is no statistically significant difference in dropout rates between male and female students.

### In Context of the Dataset

- The results suggest that gender does not significantly impact student dropout.

- Male and female students show similar dropout patterns, indicating that other factors may be more important.

---

### Hypothesis 3 – Scholarship Status and Dropout

H₀: Scholarship status and dropout are independent.

H₁: Scholarship status and dropout are associated.

Test: Chi-Square Test

In the context of the dataset: The analysis will examine whether students with a scholarship are less likely to drop out compared to those without a scholarship.

In [39]:
contingency_table = pd.crosstab(df["Scholarship"], df["Dropout"])
contingency_table

Dropout,0,1
Scholarship,,
0,4456,1391
1,2446,727


In [40]:
chi2, p_value, dof, expected = stats.chi2_contingency(contingency_table)

In [41]:
print("Chi-square statistic:", chi2)
print("P-value:", p_value)

Chi-square statistic: 0.8340947475409783
P-value: 0.36109115621155796


#### What this code does

- A contingency table was created using Scholarship and Dropout to show the frequency of each group.

- A Chi-Square test was performed to examine the relationship between scholarship status and dropout.

- The test outputs a chi-square statistic and a p-value to determine whether the variables are associated.

### Interpretation

- The test produced a chi-square statistic of 0.83 and a p-value of 0.361.

- Since the p-value is greater than 0.05, the null hypothesis is not rejected.

- This indicates that there is no statistically significant association between scholarship status and dropout.

### In Context of the Dataset

- The results suggest that having a scholarship does not significantly affect whether a student drops out.

- Students with and without scholarships show similar dropout patterns, indicating that other factors may have a stronger influence.